src/components/StatePicker.js

```diff
@@ -1,67 +1,102 @@
 import PropTypes from 'prop-types';
-import Picker from './Picker';
+import compose from '../libs/compose';
+import withNavigation from './withNavigation';
+import styles from '../styles/styles';
 import withLocalize, {withLocalizePropTypes} from './withLocalize';
+import MenuItemWithTopDescription from './MenuItemWithTopDescription';
+import Navigation from '../libs/Navigation/Navigation';
+import ROUTES from '../ROUTES';
+import FormHelpMessage from './FormHelpMessage';
 
 const propTypes = {
-    /** The label for the field */
-    label: PropTypes.string,
-
-    /** A callback method that is called when the value changes and it receives the selected value as an argument. */
-    onInputChange: PropTypes.func.isRequired,
-
-    /** The value that needs to be selected */
-    value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
-
-    /** The ID used to uniquely identify the input in a Form */
-    inputID: PropTypes.string,
-
-    /** Saves a draft of the input value when used in a form */
-    shouldSaveDraft: PropTypes.bool,
-
-    /** Callback that is called when the text input is blurred */
-    onBlur: PropTypes.func,
+    /** Current State from user address  */
+    stateISO: PropTypes.string,
 
     /** Error text to display */
     errorText: PropTypes.string,
 
+    /** Default value to display */
+    defaultValue: PropTypes.string,
+
     ...withLocalizePropTypes,
 };
 
 const defaultProps = {
-    label: '',
-    value: undefined,
+    stateISO: '',
     errorText: '',
-    shouldSaveDraft: false,
-    inputID: undefined,
-    onBlur: () => {},
+    defaultValue: '',
 };
 
-const StatePicker = forwardRef((props, ref) => {
-    const STATES = _.chain(props.translate('allStates'))
-        .sortBy(state => state.stateName.toLowerCase())
-        .map(state => ({value: state.stateISO, label: state.stateName}))
-        .value();
+function BaseStatePicker(props) {
+    const route = useRoute();
+    const stateISO = props.stateISO;
+    const [stateTitle, setStateTitle] = useState(stateISO);
+    const paramStateISO = lodashGet(route, 'params.stateISO');
+    const navigation = props.navigation;
+    const onInputChange = props.onInputChange;
+    const defaultValue = props.defaultValue;
+    const translate = props.translate;
+
+    useEffect(() => {
+        if (!paramStateISO || paramStateISO === stateTitle) {
+            return;
+        }
+
+        setStateTitle(paramStateISO);
+
+        // Needed to call onInputChange, so Form can update the validation and values
+        onInputChange(paramStateISO);
+
+        navigation.setParams({stateISO: null});
+    },
+    [paramStateISO, stateTitle, onInputChange, navigation]);
+
+    const navigateToCountrySelector = useCallback(() => {
+        Navigation.navigate(ROUTES.getUsaStateSelectionRoute(stateTitle || stateISO, Navigation.getActiveRoute()));
+    }, [stateTitle, stateISO]);
+
+    const title = useMemo(() => {
```

Comment:
I think memoization is not needed here

Reply:
I propose to write a separate function for that

Reply:
If I move it into a function I will need to pass the props, and according to the guidelines I will wrap it into a `useCallback` so will look kind the same but will recalculate the same values so I would keep it into a memo given that other components can cause re-renders

<img width="185" alt="Screen Shot 2023-05-05 at 16 55 33" src="https://user-images.githubusercontent.com/1676818/236493668-9269ed6a-1e0c-4131-8a6f-4d65f8d51c76.png">

Reply:
But this is a string type, which shouldn't cause a re-render in case the title stays the same without using useMemo, that because I requested removing that because it's a cheap function